From ddc39b45cad3ae038e21843069c90f64aa61c43c Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Thu, 31 May 2012 11:46:22 +0900
Subject: [PATCH 43/73] mediabus: add new 32 bits data formats on the media bus

This adds the following RGB data formats which mean each pixel
data is extended from 18/24 to 32 bits; V4L2_MBUS_FMT_RGB666_1X18_LE,
V4L2_MBUS_FMT_RGB888_1X24_LE, and V4L2_MBUS_FMT_RGB888_2X12_LE.
---
 drivers/media/video/soc_mediabus.c |   21 +++++++++++++++++++++
 include/media/v4l2-mediabus.h      |    3 +++
 2 files changed, 24 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/soc_mediabus.c b/drivers/media/video/soc_mediabus.c
index 9d14948..866ab8d 100644
--- a/drivers/media/video/soc_mediabus.c
+++ b/drivers/media/video/soc_mediabus.c
@@ -74,6 +74,27 @@ static const struct soc_mbus_pixelfmt mbus_fmt[] = {
 		.packing		= SOC_MBUS_PACKING_2X8_PADHI,
 		.order			= SOC_MBUS_ORDER_LE,
 	},
+	[MBUS_IDX(RGB666_1X18_LE)] = {
+		.fourcc			= V4L2_PIX_FMT_RGB32,
+		.name			= "RGB666/32bpp",
+		.bits_per_sample	= 18,
+		.packing		= SOC_MBUS_PACKING_EXTEND32,
+		.order			= SOC_MBUS_ORDER_LE,
+	},
+	[MBUS_IDX(RGB888_1X24_LE)] = {
+		.fourcc			= V4L2_PIX_FMT_RGB32,
+		.name			= "RGB888/32bpp",
+		.bits_per_sample	= 24,
+		.packing		= SOC_MBUS_PACKING_EXTEND32,
+		.order			= SOC_MBUS_ORDER_LE,
+	},
+	[MBUS_IDX(RGB888_2X12_LE)] = {
+		.fourcc			= V4L2_PIX_FMT_RGB32,
+		.name			= "RGB888/32bpp",
+		.bits_per_sample	= 24,
+		.packing		= SOC_MBUS_PACKING_EXTEND32,
+		.order			= SOC_MBUS_ORDER_LE,
+	},
 	[MBUS_IDX(SBGGR8_1X8)] = {
 		.fourcc			= V4L2_PIX_FMT_SBGGR8,
 		.name			= "Bayer 8 BGGR",
diff --git a/include/media/v4l2-mediabus.h b/include/media/v4l2-mediabus.h
index 865cda7..9b189b4 100644
--- a/include/media/v4l2-mediabus.h
+++ b/include/media/v4l2-mediabus.h
@@ -32,6 +32,9 @@ enum v4l2_mbus_pixelcode {
 	V4L2_MBUS_FMT_RGB555_2X8_PADHI_BE,
 	V4L2_MBUS_FMT_RGB565_2X8_LE,
 	V4L2_MBUS_FMT_RGB565_2X8_BE,
+	V4L2_MBUS_FMT_RGB666_1X18_LE,
+	V4L2_MBUS_FMT_RGB888_1X24_LE,
+	V4L2_MBUS_FMT_RGB888_2X12_LE,
 	V4L2_MBUS_FMT_SBGGR8_1X8,
 	V4L2_MBUS_FMT_SBGGR10_1X10,
 	V4L2_MBUS_FMT_GREY8_1X8,
-- 
1.7.0.4

